soc/nordic/nrf54l/Kconfig

```diff
@@ -56,8 +67,15 @@ config SOC_NRF_FORCE_CONSTLAT
 
 config SOC_NRF54L_VREG_MAIN_DCDC
 	bool "NRF54L DC/DC converter."
+	select DEPRECATED
 	help
 	  To enable, an inductor must be connected to the DC/DC converter pin.
+	  This option is deprecated, use devicetree instead. Example
+	  configuration:
+
+	  &vregmain {
+	    regulator-initial-mode = <NRF5X_REG_MODE_DCDC>;
+	  };
```

Comment:
I don't think there's a need to deprecate on nRF54L (still in engineering phase)

Reply:
I updated code with dts-like approach

Reply:
I will update L15 in another PR, to do not mix two tasks in one PR

Reply:
see: https://github.com/zephyrproject-rtos/zephyr/pull/77160